kernel.go

```diff
@@ -373,6 +369,62 @@ func handleExecuteRequest(ir *classic.Interp, receipt msgReceipt) error {
 	return receipt.Reply("execute_reply", content)
 }
 
+// doEval evaluates the code in the interpreter. This function captures an uncaught panic
+// as well as the value of the last statement/expression.
+func doEval(ir *classic.Interp, code string) (val interface{}, err error) {
+	// Capture a panic from the evaluation if one occurs
+	defer func() {
+		if r := recover(); r != nil {
+			var ok bool
+			if err, ok = r.(error); !ok {
+				err = errors.New(fmt.Sprint(r))
+			}
+		}
+	}()
+
+	// Prepare and perform the multiline evaluation.
+	env := ir.Env
+	env.Options &^= base.OptShowPrompt
+	env.Options &^= base.OptTrapPanic
+	env.Line = 0
+
+	// Parse the input code (and don't preform gomacro's macroexpansion).
+	src := ir.ParseOnly(code)
+
+	// Check if the last node is an expression.
+	var srcEndsWithExpr bool
+	if src != nil {
+		if srcAstWithNode, ok := src.(ast2.AstWithNode); ok {
```

Comment:
Are these the only two options here?  If not, maybe a switch with a default would be useful.  I so, I don't think the `else if` is needed.  I prefer to stay away from `else` statements.  They are generally not needed in Go.

Also, what happens if `src == nil`? Should it still go to `Eval`?  If not, you could just have an if statement above this block that returns if `src == nil`.

Reply:
These are the only 2 options (well `nil` is a third but handled with an early return as you mentioned). The goal of this is to determine if the last eval result should be returned or swallowed. Since the 2 types that are checked are not documented there is the possibility for them to change without notice and so I think this should default to skipping the print in the case where something new is returned. What do you think?

Reply:
After implementing a type switch it seems better to leave the if-else if since both branches need the src cast to the type being checked anyways.

Reply:
Ok, we can keep it for now. I might play around with it to see if there is some other way, but not too concerned if you have a justification for it.